In [1]:
import pandas as pd
import numpy as np

In [16]:
df=pd.read_csv(r"C:\Users\shreyaash mogaveera\sentiment-analysis-nlp\data\IMDB Dataset.csv")

In [17]:
df.head()

,review,sentiment
0,One of the other reviewers has mentioned that ...,positive
1,A wonderful little production. <br /><br />The...,positive
2,I thought this was a wonderful way to spend ti...,positive
3,Basically there's a family where a little boy ...,negative
4,"Petter Mattei's ""Love in the Time of Money"" is...",positive


In [27]:
df['review'][4]

'Petter Mattei\'s "Love in the Time of Money" is a visually stunning film to watch. Mr. Mattei offers us a vivid portrait about human relations. This is a movie that seems to be telling us what money, power and success do to people in the different situations we encounter. <br /><br />This being a variation on the Arthur Schnitzler\'s play about the same theme, the director transfers the action to the present time New York where all these different characters meet and connect. Each one is connected in one way, or another to the next person, but no one seems to know the previous point of contact. Stylishly, the film has a sophisticated luxurious look. We are taken to see how these people live and the world they live in their own habitat.<br /><br />The only thing one gets out of all these souls in the picture is the different stages of loneliness each one inhabits. A big city is not exactly the best place in which human relations find sincere fulfillment, as one discerns is the case wit

In [18]:
df.shape

(50000, 2)

In [19]:
df['sentiment'].value_counts()

sentiment
positive    25000
negative    25000
Name: count, dtype: int64

In [22]:
df['review_length']=df['review'].apply(len)
df['review_length'].mean()

np.float64(1309.43102)

In [23]:
df.isnull().sum()

review           0
sentiment        0
review_length    0
dtype: int64

In [24]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 3 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   review         50000 non-null  object
 1   sentiment      50000 non-null  object
 2   review_length  50000 non-null  int64 
dtypes: int64(1), object(2)
memory usage: 1.1+ MB


In [26]:
df.duplicated().sum()

np.int64(418)

## Key insights
 - dataset contains 50000 rows n 2 columns
 - HTML tags are present in reviews
 - data is perfectly balanced between positive and negative
 - average: ~1300 characters
 - there are no missing values but it contains 418 duplicate data